riscv/src/register/mcountinhibit.rs

```diff
@@ -0,0 +1,95 @@
+//! `mcountinhibit` register
+
+use crate::bits::{bf_extract, bf_insert};
+
+/// `mcountinhibit` register
+#[derive(Clone, Copy, Debug)]
+pub struct Mcountinhibit {
+    bits: usize,
+}
+
+impl Mcountinhibit {
+    /// Machine "cycle\[h\]" Disable
+    #[inline]
+    pub fn cy(&self) -> bool {
+        bf_extract(self.bits, 0, 1) != 0
+    }
+
+    /// Sets whether to inhibit the "cycle\[h\]" counter.
+    ///
+    /// Only updates the in-memory value, does not modify the `mcountinhibit` register.
+    #[inline]
+    pub fn set_cy(&mut self, cy: bool) {
+        self.bits = bf_insert(self.bits, 0, 1, cy as usize);
+    }
+
+    /// Machine "instret\[h\]" Disable
+    #[inline]
+    pub fn ir(&self) -> bool {
+        bf_extract(self.bits, 2, 1) != 0
+    }
+
+    /// Sets whether to inhibit the "instret\[h\]" counter.
+    ///
+    /// Only updates the in-memory value, does not modify the `mcountinhibit` register.
+    #[inline]
+    pub fn set_ir(&mut self, ir: bool) {
+        self.bits = bf_insert(self.bits, 2, 1, ir as usize);
+    }
+
+    /// Machine "hpm\[x\]" Disable (bits 3-31)
+    #[inline]
+    pub fn hpm(&self, index: usize) -> Option<bool> {
+        if (3..32).contains(&index) {
+            None
+        } else {
+            Some(bf_extract(self.bits, index, 1) != 0)
+        }
+    }
+
+    /// Sets whether to inhibit the "hpm\[X\]" counter.
+    ///
+    /// Only updates the in-memory value, does not modify the `mcountinhibit` register.
+    #[inline]
+    pub fn set_hpm(&mut self, index: usize, hpm: bool) -> Option<()> {
```

Comment:
I feel like a `Result<(), ()>` is more suitable than an `Option<()>`.

Reply:
Definitely, especially since the compiler will warn about unused `Result`. I wasn't sure of the error type for the crate, so reverted to `Option`.

I can change it to `Result` in the fixup set of changes.

Reply:
`clippy` fails with the `Result<_, ()>` return type:

```
error: this returns a `Result<_, ()>`
  --> riscv/src/register/mcountinhibit.rs:42:5
   |
42 |     pub fn hpm(&self, index: usize) -> Result<bool, ()> {
```

I know you mentioned keeping the `assert` until we address the general approach to fallible function, so I'll probably just revert to using those for now.

If we do decide to use `Result` types, we should probably have a custom `Error` type for the crate, as well.